src/cuga/backend/cuga_graph/nodes/cuga_supervisor/cuga_supervisor_graph.py

```
@@ -512,19 +548,44 @@ async def call_model(state: CugaSupervisorState, config: Optional[RunnableConfig
 
             logger.debug(f"Total messages for model (including system): {len(messages_for_model)}")
 
-            response = await base_model.ainvoke(messages_for_model, config=config or {})
+            bind_tools_list = (tools_for_llm_ref or {}).get("bind_tools_list") if tools_for_llm_ref else None
+            use_function_calling = bool(bind_tools_list)
+
+            invoke_model = base_model
+            if use_function_calling:
+                try:
+                    invoke_model = base_model.bind_tools(bind_tools_list)
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)